internal/js_scanner/js_scanner.go

```diff
@@ -339,7 +339,7 @@ outer:
 		// Track our current position
 		i += len(value)
 	}
-	if len(genericsIdents) > 0 && ident != defaultPropType {
+	if start > 0 && len(genericsIdents) > 0 && ident != defaultPropType {
```

Comment:
heh? is this just saying that `Props` is a magic name

Reply:
Yep, `Props` is a magic name. In this case our lexing was getting tripped up because of the `Record<string, never>` and it was injecting the whole `---
type Props = Record<string, never>` string in the wrong place.